scrapy_poet/po_tester.py

```diff
@@ -0,0 +1,78 @@
+from hashlib import md5
+from pathlib import Path
+from typing import Callable, Mapping
+
+from scrapy import Request, Spider, signals
+from scrapy.crawler import Crawler
+from scrapy.http import Response
+from scrapy.utils.project import get_project_settings
+
+from scrapy_poet import InjectionMiddleware
+from scrapy_poet.injection import Injector
+from scrapy_poet.overrides import OverridesRegistry
+from url_matcher.util import get_domain
+from twisted.internet.defer import inlineCallbacks
+
+
+class DummySpider(Spider):
+    name = "dummy_spider"
+
+
+def get_injector(additional_settings: Mapping) -> Injector:
+    settings = get_project_settings()
+    for k, v in additional_settings.items():
+        settings.set(k, v)
+    crawler = Crawler(DummySpider)
+    crawler.settings = settings
+    spider = DummySpider()
+    spider.settings = settings
+    crawler.spider = spider
+    injector_middleware = InjectionMiddleware.from_crawler(crawler)
+    return injector_middleware.injector
+
+
+def get_replaying_injector(fixture_path: Path) -> Injector:
+    settings = dict(
+        SCRAPY_POET_CACHE=str(fixture_path.absolute()),
+    )
+    return get_injector(settings)
+
+
+class POTester:
+    """
+    Reply Page Objects inputs so that unit tests can be created easily
+    """
+
+    def __init__(self, url: str, po_type: Callable, tests_path: Path):
+        self.url = url
+        self.po_type = po_type
+        self.domain = get_domain(url)
+        self.tests_root_path = tests_path
+        self.tests_root_path.mkdir(parents=True, exist_ok=True)
+        self.fixtures_root_path = self.tests_root_path / "fixtures"
+        self.fixtures_root_path.mkdir(parents=True, exist_ok=True)
+
+        # Hash using the po_type and the url
+        hash = md5(f"{po_type.__module__}.{po_type.__name__}/{url}".encode()).hexdigest()[:8]
```

Comment:
Is it ok to use built-in `hash` name here?

Reply:
Using `md5` is better as it's in 128-bit space than using the built-in `hash` which operates in a 64-bit space. We'd probably not have a collision entry unless we'd have billions of Page Objects. 😅 Nonetheless, I believe we should `md5` given its slight benefits.